package.json

```diff
@@ -17,10 +17,7 @@
 	],
 	"icon": "img/vscode-lotro-api.png",
 	"activationEvents": [
-		"onLanguage:plugin",
-		"workspaceContains:**/SkinDefinition.xml",
-		"workspaceContains:**/*.plugin",
```

Comment:
the workspaceContains:**/*.plugin is still needed so as to prevent the entire exension from loading for non lotro projects that happen to contain xml.

Reply:
The issue with workspaceContains is that it doesn't work if the file being opened isn't part of the workspace folder. For example, I could be working on OneLauncher and want to edit a compendium file from the documents dir to test something. In that case, relying on workspaceContains wouldn't work, as it only has my Python stuff.

Would it be possible to just run some cheap code before the extension is fully activated to verify that an open file matches something the extension supports?

Reply:
There's also the possibility of making a custom language or languages for the supported files (checking for filenames like SkinDefinition.xml can be done too), and then just setting the language to xml in the extension once it is activated with `setTextDocumentLanguage`.